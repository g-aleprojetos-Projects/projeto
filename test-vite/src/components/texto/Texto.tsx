
import textos from '../../resourses/textos';
import * as S from './Texto.styles';

export type Props = {
  id?: string;
  cor?: string;
  tamanho?: number;
  alturaDeLinha?: number;
  marginTop?: number;
  children?: string | React.ReactNode;
  peso?: 'bold' | 'medium';
  toUpper?: boolean;
  cursor?: 'default' | 'pointer' | 'text';
};

export function Texto(props: Props) {
  return (
    <S.TextoH1
      tamanho={textos.tamanho.big}
      alturaDeLinha={textos.altura.big}
      {...props}
    />
  );
}

