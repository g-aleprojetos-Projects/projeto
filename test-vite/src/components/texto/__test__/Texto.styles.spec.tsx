import renderer from 'react-test-renderer';
import 'jest-styled-components';
import * as S from '../Texto.styles';
import colors from '../../../resourses/colors';

describe('Texto.styles', () => {

  test('TextoH1 DEVE ser igual ao snapshot', () => {
    const texto = renderer.create(<S.TextoH1>Teste</S.TextoH1>).toJSON();

    expect(texto).toMatchSnapshot();
  });

  test(`TextoH1 DEVE ter a cor igual a "##000000" QUANDO a propriedade cor informada for "${colors.black}"`, () => {
    const texto = renderer
      .create(<S.TextoH1 cor={colors.black}>Teste</S.TextoH1>)
      .toJSON();
    console.log(texto)
    expect(texto).toHaveStyleRule('color', '#000000');
  });
});
