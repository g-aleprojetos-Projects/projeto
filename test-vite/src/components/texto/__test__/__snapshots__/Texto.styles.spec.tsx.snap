// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Texto.styles > TextoH1 DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-family: AvertaStd-Regular;
  color: #ffffff;
  margin-top: 0px;
  text-transform: none;
  cursor: text;
}

<h1
  className="c0"
>
  Teste
</h1>
`;
